ci/check-crates.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+# env:
+#   - CRATE_TOKEN
+
+declare skip_patterns=(
```

Comment:
I used the pattern list because I guess we will ignore some normal Cargo.toml. (like `./programs/sbf/rust/**/Cargo.toml`)

seems it relates to what we got in next line 🤔

Reply:
I think we can just skip workspaces (and crates marked `package.publish=false`) if we use the command i posted on discord to resolve metadata.

one complication i realized is that we won't get the fancy workspace attribute inheritance stuff from recent cargo. for example the SBF test crates are all marked for no publication in the workspace declaration like this

```
[workspace]
members = [ ... ]

[workspace.package]
publish = false
```

so the individual `Cargo.toml` don't declare it.  not insurmountable, but a complication for sure.  it'd be nice if there were a cargo tool that fully resolved the crate definition and dumped the aggregate :pensive: